### PR TITLE
QueuePeeker does select with READPAST hint

### PR DESCRIPTION
**Pull request by tmasternak**
Connects to #248

### CODE REVIEW

**Comment by tmasternak** [comment]
ping @Particular/sqlserver-transport-maintainers this is ready to review

**Comment by tmasternak** [comment]
thx @HEskandari :tada:

File: src/NServiceBus.SqlServer/Queuing/TableBasedQueue.cs [hunk 1]
```diff
@@ -18,11 +18,11 @@ public TableBasedQueue(QueueAddress address)
 
         public string TransportAddress => address.ToString();
 
-        public virtual async Task<int> TryPeek(SqlConnection connection, CancellationToken token)
+        public virtual async Task<int> TryPeek(SqlConnection connection, CancellationToken token, int timeoutInSeconds = 30)
         {
             var commandText = Format(Sql.PeekText, address.SchemaName, address.TableName);
 
-            using (var command = new SqlCommand(commandText, connection))
+            using (var command = new SqlCommand(commandText, connection) {CommandTimeout = timeoutInSeconds})
```
Comment (danielmarbach) [inline src/NServiceBus.SqlServer/Queuing/TableBasedQueue.cs]: @tmasternak seems to have a formatting issue here

Comment (tmasternak) [inline src/NServiceBus.SqlServer/Queuing/TableBasedQueue.cs]: @danielmarbach do you mean `CommandTimeout` initialization part?

Comment (danielmarbach) [inline src/NServiceBus.SqlServer/Queuing/TableBasedQueue.cs]: Yes

**Pull request merged**